zwave_js_server/model/node.py

```diff
@@ -454,6 +485,11 @@ def handle_value_removed(self, event: Event) -> None:
     def handle_value_notification(self, event: Event) -> None:
         """Process a node value notification event."""
         # append metadata if value metadata is available
+        # If endpoint isn't provided, we can assume the endpoint is 0
+        # https://zwave-js.github.io/node-zwave-js/#/api/valueid
+        if event.data["args"].get("endpoint") is None:
+            event.data["args"]["endpoint"] = 0
+
         value = self.values.get(_get_value_id_from_dict(self, event.data["args"]))
```

Comment:
Where are we storing ValueNotification instances in `Node.values`?

Reply:
We are not storing ValueNotification instances. They get attached to the value notification event by the library before getting passed on to the consumer, e.g. the `zwave_js` integration.

Reply:
I think what's confusing me is that we're casting the Value instance to a ValueNotification instance. That's not correct. We should always create a ValueNotification instance with the same data as the Value that we're basing the ValueNotification on.

Reply:
@marcelveldt do you have any thoughts on this? I can make this change to create a new instance, just not sure if it was an intentional choice to update the original Value rather than creating a new ValueNotification instance?

Reply:
You need to fetch the metadata belonging to the Value notification, this will be sent upfront as "value added", "value updated" or "metadata updated" message. We could also revert to the previous behavior where the server collected the metadata on an incoming value notification message...

Reply:
That part I understand. It's a matter of order of operations and what object is being updated though. Right now, we are updating the `Value` instance that we retrieve with the value notification data. This means that we are updating the state of the `Value` we are storing. Is this intentional? The alternative approach is to create a new `ValueNotification` instance from the `Value` and update it with the value notification data. This achieves the same thing but doesn't impact the state of the original `Value`

Reply:
Regardless of the question about updating the existing `Value` instance we should always create a `ValueNotification` instance and set it in the event data. We can't mix types for the same event data.

Reply:
`Value` and `ValueNotification` are the same thing internally. Is there a functional issue or is this more of a "right way to do this"?

Reply:
It was just a bit future proof at that time but it doesn't make any actual sense to keep it if we're not going to add additional logic.

Reply:
> That part I understand. It's a matter of order of operations and what object is being updated though. Right now, we are updating the `Value` instance that we retrieve with the value notification data. This means that we are updating the state of the `Value` we are storing. Is this intentional? The alternative approach is to create a new `ValueNotification` instance from the `Value` and update it with the value notification data. This achieves the same thing but doesn't impact the state of the original `Value`

Uhm, we could probably request the metadata at this point (there's a function for that) and update the value in our states dict. So that way we can guarantee we always have the metadata. That would solve this once and for all.

value notification --> is this value known: use known metadata. value not known and no metadata in notification message, request the metadata from the server, create/update value object and send the complete result to HA

Reply:
> `Value` and `ValueNotification` are the same thing internally. Is there a functional issue or is this more of a "right way to do this"?

`ValueNotification` inherits `Value`, but it doesn't mean we can replace a `ValueNotification` instance with a `Value` instance, even if they have the same features currently.

Reply:
I see what you mean. I can submit a separate PR to address that